modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/spi/model/query/contributor/DefaultKeywordQueryContributor.java

```diff
@@ -76,13 +83,47 @@ public void contribute(
 		}
 	}
 
+	public int getMaximumKeywordSize() {
+		return _maximumKeywordSize;
+	}
+
+	public String trimKeywords(String keywords) {
+		if (keywords.length() < _maximumKeywordSize) {
```

Comment:
Null check :bug:

Reply:
Also, we probably not want to limit the number of keywords, just the length of an individual keyword so this check is not needed.

Reply:
For the vast majority of cases we expect search terms to be below the _maximumKeywordSize, and String manipulation is somewhat expensive, I think a fast return here is worthwhile.